README.md

```diff
@@ -51,6 +51,132 @@ After this, you will have a directory containing files used for creating a
+
+        - **`index.js`**\
+Webpack bundler entry point for generating front-end code bundle.
+
+        - **`labplugin.js`**\
+JupyterLab extension definition for JupyterLab integration. When JpuyterLab loads, it activates the plugin defined in this file. Widget registry service provided by `@jupyter-widgets/jupyterlab-manager` is used during activation to register the widget with JupyterLab.
+
+    - **`package.json`**\
+npm package definiton file for widget front-end
+
+    - **`README.md`**\
+Auto-generated documentation file for front-end npm package, with installation instructions.
+
+    - **`webpack.config.js`**\
+Webpack bundler configuration that defines bundler entry points and ouputs along with other configuration
+
+- **`.gitignore`**\
+Ignore rules for widget project git repository.
+
+- **`<npm_package_name>.json`**\
+nbextension configuration file used by Jupyter Notebook
+
+- **`MANIFEST.in`**\
+setuptools configuration file containing commands to execute (include / exclude files) during installation (pip install).
+
+- **`README.md`**\
+Auto-generated documentation file for widget project.
+
+- **`RELEASE.md`**\
+Auto-generated documentation file for publishing python package to PyPI and npm package to npm registry.
+
+- **`setup.cfg`**\
+Configuration file to mark the python package as a Universal Wheel.
+
+- **`setup.py`**\
+Build script for the widget package.
+
+## Local Dev Installation
+- `pip install -e .`
+- `jupyter nbextension install --py --symlink --sys-prefix <python_package_name>`
+- `jupyter nbextension enable --py --sys-prefix <python_package_name>`
+- `jupyter labextension install js`
+- After making python code changes
+    - Restarting kernel will be enough to reflect changes
+- After making javascript code changes
+    - `cd js`
+    - `npm run build`
+    - Refresh browser to reflect changes
+
+## Publishing Widget
+Follow the steps in (RELEASE.md) to publish widget python package and npm packages.
+
+## Widget Installation Process for Notebook
+- `pip install` installs python package for access from kernel
+- `nbextension install` copies front-end code files from `/static` directory into `<jupyter path>/nbextensions/<extension_name>`
+
+## Widget Load Process for Notebook
+- Python code is loaded into kernel from python package directory
+- Front-end javascript code loaded from `<jupyter path>/nbextensions/<extension_name>/extension.js`
+
+## Widget Installation Process for Jupyter Lab
+- Notebook installation as above first
+- `labextension install` copies npm package into JupyterLab extension staging folder
+- Rebuilding JupyterLab integrates widget npm package code into JupyterLab bundle
+
+## Widget Load Process for Jupyter Lab
+- Python code is loaded into kernel from python package directory as in notebook
+- Front-end javascript code loaded from `labplugin.js` of npm package
```

Comment:
Not sure this is fully correct. Really, `jupyterlab build` rebuilds the full JupyterLab frontend.